plasoscaffolder/plugins/manager.py

```diff
@@ -0,0 +1,163 @@
+# -*- coding: utf-8 -*-
+"""The plugin manager."""
+
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Generator
+from typing import Tuple
+from typing import Type
+
+from plasoscaffolder.plugins import interface
+
+
+class PluginManager(object):
+  """The plugin manager."""
+
+  _plugin_classes = {}
+
+  @classmethod
+  def DeregisterPlugin(cls, plugin_class: Type[interface.ScaffolderPlugin]):
+    """Deregisters a plugin class.
+
+    The plugin classes are identified based on their lower case name.
+
+    Args:
+      plugin_class (type): plugin class (subclass of ScaffolderPlugin).
+
+    Raises:
+      KeyError: if plugin class is not set for the corresponding name.
+    """
+    plugin_provides = plugin_class.PROVIDES.lower()
```

Comment:
Is PROVIDES the name of the plugin? If so, please rename this constant in the interface.